### PR TITLE
Simplify env

### PR DESCRIPTION
**Pull request by brandonchinn178**
After #89, the stubs libraries are put in PYTHONPATH, but other external deps are still in MYPYPATH. Now, we can just put everything into PYTHONPATH, and remove the logic to skip adding  external deps if the stub library is added, since mypy already has the logic to load `types-foo` over `foo`.

Also includes refactoring the MYPYPATH logic to be simpler + more readable. There are probably edge cases that are dropped in the refactoring, but I think we can find regressions + add test cases as we go

### CODE REVIEW

File: mypy/private/mypy.bzl [hunk 1]
```diff
@@ -61,6 +40,81 @@ def _opt_in(opt_in_tags, rule_tags):
...
+    # output bin_dir from `generated_dirs` first
+    # https://github.com/theoremlp/rules_mypy/issues/88
+    mypy_path = ["."]
+
+    # generated files intended to be imported as absolute paths
+    mypy_path.append(ctx.bin_dir.path)
+
+    for imp in ctx.rule.attr.imports:
+        # importing files in same target
+        mypy_path.append("{}/{}".format(ctx.label.package, _get_main_relpath(imp)))
```
Comment (keith) [inline mypy/private/mypy.bzl:96]: if this function returns None it adds an invalid path here
